src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/CompilationTask.kt

```diff
@@ -499,6 +506,27 @@ private fun JvmCompilationTask.updateBuilder(
     it.build()
   }
 
+/**
+ * Copy generated manifest files from KSP task into generated folder
+ */
+internal fun Iterator<String>.copyManifestFilesToGeneratedClasses(directories: Directories): Iterator<String> {
+    val result = mutableSetOf<String>()
+    this.forEach {
+        if ("/$MANIFEST_DIR" in it) {
+            val path = Paths.get(it)
+            val srcJarsPath = Paths.get(directories.temp, SOURCE_JARS_DIR)
+            if (srcJarsPath.exists()) {
+                val relativePath = srcJarsPath.relativize(path)
+                val destPath = Paths.get(directories.generatedClasses).resolve(relativePath)
+                destPath.parent.toFile().mkdirs()
+                Files.copy(path, destPath, StandardCopyOption.REPLACE_EXISTING)
```

Comment:
And same here, is this actually needed?

Reply:
same as this https://github.com/bazelbuild/rules_kotlin/pull/1094/#discussion_r1430697819